### PR TITLE
PAYARA-4065 Ensure that the threadpools are correctly initialized even when Embedded server is restarted

### PR DESCRIPTION
**Pull request by realityforge**
While there was a previous attempt to do this in place, it failed if the shutdown of the thread pool took any amount of time. And thus calling `stop()` and then `start()` in rapid succession would fail. This change attempts to eliminate this scenario

### CODE REVIEW

File: nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java [hunk 1]
```diff
@@ -131,14 +129,29 @@ public void postConstruct() {
...
-            // there is awkward point in embedded EJBContainer initialization, where same server
-            // instance is started twice. In such case, we need to reinitialize.
-            if (threadPoolExecutor.isShutdown()) {
-                initialiseThreadPools();
-            }
+            // Embedded containers can be started and stopped multiple times.
+            // Thus we need to initialize anytime the server instance is started.
+            initialiseThreadPools();
         } else if (event.is(EventTypes.SERVER_SHUTDOWN)) {
             threadPoolExecutor.shutdown();
```
Comment (svendiedrichsen) [inline nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java]: Maybe have a method like `terminateThreadPools` for this?

Comment (realityforge) [reply nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java]: Seems reasonable ... and done.

**Comment by pdudits** [comment]
jenkins test please

**Comment by pdudits** [comment]
jenkins test please

File: nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java [hunk 2]
```diff
@@ -120,8 +120,6 @@ public void postConstruct() {
         if (transactions != null) {
             transactions.addListenerForType(PayaraExecutorServiceConfiguration.class, this);
         }
-
-        initialiseThreadPools();
```
Comment (pdudits) [inline nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java]: I believe it is this point that makes the tests fail. Reinitializing at ALL_APPLICATIONS_LOADED was a heuristic I found to work for multiple embedded starts, but the threadpool is already needed during load.

File: nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java [hunk 3]
```diff
@@ -131,15 +129,34 @@ public void postConstruct() {
...
     public void event(Event event) {
         if (event.is(Deployment.ALL_APPLICATIONS_LOADED)) {
-            // there is awkward point in embedded EJBContainer initialization, where same server
-            // instance is started twice. In such case, we need to reinitialize.
-            if (threadPoolExecutor.isShutdown()) {
-                initialiseThreadPools();
-            }
+            // Embedded containers can be started and stopped multiple times.
+            // Thus we need to initialize anytime the server instance is started.
+            initialiseThreadPools();
```
Comment (pdudits) [inline nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java]: And in such case the test whether he executor is alive is needed.

File: nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java (hunk 2 above)
Comment (realityforge) [reply nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java]: Seems reasonable. I have reworked it slightly to follow the same strategy but use null value of pool as the trigger to recreate

**Comment by pdudits** [comment]
jenkins test please

**Pull request merged**